components/mediation/security/org.wso2.micro.integrator.mediation.security/src/main/java/org/wso2/micro/integrator/mediation/security/vault/SecretCipherHander.java

```diff
@@ -46,4 +80,31 @@ String getSecret(String alias) {
 		return parentRepository.getSecret(alias);
 	}
 
-}
\ No newline at end of file
+	public String getSecret(String alias, SecretSrcData secretSrcData) {
+		switch (secretSrcData.getVaultType()) {
+			case DOCKER:
```

Comment:
What if we store a map of resolvers for each type and just call map.get(Type).getSecret(). Wouldn't the method look much cleaner that way?

Reply:
fixed as discussed